database.go

```diff
@@ -97,11 +105,37 @@ type databaseFormat struct {
 	Time  time.Time
 }
 
+// ErrDBNotReady is a public error type with a wait method for database.
+type ErrDBNotReady struct {
```

Comment:
Nit: Go style is to name error *types* as DBNotReadyError and *variables* as errDBNotReady